src/main/java/com/pmm/budgetcalculator/expensing/entity/mapper/ExpenseCategoryMapper.java

```diff
@@ -0,0 +1,21 @@
+package com.pmm.budgetcalculator.expensing.entity.mapper;
+
+import com.pmm.budgetcalculator.expensing.entity.Expense;
+import com.pmm.budgetcalculator.expensing.entity.ExpenseCategory;
+import com.pmm.budgetcalculator.expensing.entity.dto.ExpenseCategoryDto;
+import org.mapstruct.Mapper;
+
+import java.util.List;
+@Mapper(componentModel = "spring")
+public interface ExpenseCategoryMapper  {
+
+    ExpenseCategory dtoToEntity(ExpenseCategoryDto expenseCategoryDto);
+
+    ExpenseCategoryDto entityToDto(ExpenseCategory expenseCategory);
+
+    ExpenseCategoryDto entityToDto(Expense expense);
+
+    List<ExpenseCategoryDto> entityListToDtoList(List<ExpenseCategory> expenseCategoryList);
```

Comment:
Mapping list to listDto is redundant. Iterating on list and mapping each element is acceptable and way to go